task-native/src/main/java/org/ballerinalang/stdlib/task/actions/TaskActions.java

```diff
@@ -42,94 +38,93 @@
         }
         return null;
     }
 
     public static Object start(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
-        TaskServerConnector serverConnector = new TaskServerConnectorImpl(task);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            serverConnector.start();
-        } catch (SchedulingException e) {
-            return createTaskError(e.getMessage());
+            taskScheduler.start();
+        } catch (SchedulerException e) {
+            return Utils.createTaskError("Cannot start Listener/Scheduler." + e.getMessage());
         }
         return null;
     }
 
     public static Object stop(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
-        TaskServerConnector serverConnector = new TaskServerConnectorImpl(task);
+        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            serverConnector.stop();
-        } catch (SchedulingException e) {
-            return createTaskError(e.getMessage());
+            taskScheduler.stop(triggerID);
+        } catch (SchedulerException e) {
+            return Utils.createTaskError("Cannot stop Listener/Scheduler." + e.getMessage());
         }
         return null;
     }
 
     public static Object attach(BObject taskListener, BObject service, Object... attachments) {
+        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         ServiceInformation serviceInformation;
         if (attachments == null) {
             serviceInformation = new ServiceInformation(Runtime.getCurrentRuntime(), service);
         } else {
             serviceInformation = new ServiceInformation(Runtime.getCurrentRuntime(), service, attachments);
         }
-
-        /*
-         * TODO: After #14148 fixed, use compiler plugin to validate the service
-         */
         try {
-            validateService(serviceInformation);
+            Utils.validateService(serviceInformation);
+            BMap<BString, Object> configurations = taskListener.getMapValue(TaskConstants.
+                    MEMBER_LISTENER_CONFIGURATION);
+            taskScheduler.addService(serviceInformation, configurations, triggerID);
         } catch (SchedulingException e) {
-            return createTaskError(e.getMessage());
+            return Utils.createTaskError(e.getMessage());
+        } catch (SchedulerException e) {
+            return Utils.createTaskError("Cannot attach the service to the Listener/Scheduler." + e.getMessage());
```

Comment:
```suggestion
            return Utils.createTaskError("Cannot attach the service to the Listener/Scheduler. " + e.getMessage());
```

These error messages seem inconsistent. Check for the other places as well.